presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcConfig.java

```diff
@@ -49,6 +50,18 @@ public BaseJdbcConfig setConnectionUrl(String connectionUrl)
         return this;
     }
 
+    @AssertTrue(message = "Invalid JDBC URL: has not jdbc:<subprotocol>:<subname> format.")
+    public boolean isConnectionUrlValidJdbcUri()
+    {
+        String connectionUrl = getConnectionUrl();
+        // Per JDBC 4.3 section 9.4 JDBC url must follow format jdbc:<protocol>:<subname>
+        if (connectionUrl.matches("^jdbc:([a-zA-Z0-9]+):.*?$")) {
```

Comment:
- `.*` doesn't match everything. While new lines are probably invalid, let's not validate that. you can use `(?s:.*)` to match everything, or you can use `javax.validation.constraints.Pattern(flag=...)` for that
- the braces are redundant
- the `?` is redundant

```suggestion
        if (connectionUrl.matches("^jdbc:[a-zA-Z0-9]+:(?s:.*)$")) {
```